### ISSUE TITLE
Plugin API Support

### ISSUE THREAD

**Issue by geekdave** [es-analysis/plato#29]
This project is awesome!  Just wondering what you'd think of some kind of API mechanism, so that others can crete Plato plugins.  I've seen in other issue threads that you've entertained the idea of code coverage reports, but the actual computing of the statistics should be handled outside of Plato.  With a plugin API of some sort, people could create beautiful Plato reports for just about anything--code coverage, unit test failures, developer contribution statistics, etc.

Perhaps you could have a config file for registering custom plugins like this:

``` js
plugins: {
  coverage: {
    command: "grunt coverage",
    expectedResult: "coverage-report.json"
  }
}
```

So basically Plato would spawn 3rd party scripts/processes to "do something", and we would expect those scripts/processes to output their results in some kind of well-known JSON format, which Plato would then use to create a chart.

For instance, maybe the JSON output for a code coverage report could look like this:

``` js
{
  "reportName": "Code Coverage",
  "chartType": "lineChart",
  "yAxis": "percent",
  "xAxis": "currentDate",
  "value": "72%"
}
```

This could be fed into the standard reporting mechanism that Plato uses to create historical reports over time.  So in the above example, we could see the code coverage percentage trending up or down over time.
